tests/integration/component.go

```diff
@@ -386,8 +386,13 @@ func componentTests(args ...string) {
 
 			It("should not fail when --context is not set", func() {
 				// Was failing due to https://github.com/openshift/odo/issues/1969
-				helper.Cmd("odo", append(args, "create", "--s2i", "java:8", cmpName, "--project",
-					commonVar.Project, "--binary", filepath.Join(commonVar.Context, "sb.jar"))...).ShouldPass()
+				if runtime.GOOS == "darwin" {
+					helper.Cmd("odo", append(args, "create", "--s2i", "java:8", cmpName, "--project",
+						commonVar.Project, "--binary", filepath.Join("/private", commonVar.Context, "sb.jar"))...).ShouldPass()
+				} else {
+					helper.Cmd("odo", append(args, "create", "--s2i", "java:8", cmpName, "--project",
+						commonVar.Project, "--binary", filepath.Join(commonVar.Context, "sb.jar"))...).ShouldPass()
+				}
```

Comment:
Perhaps just put the filepath part in if-else logic? Something like this? -
```go
binaryFilePath = filepath.Join(commonVar.Context, "sb.jar"))
if runtime.GOOS == "darwin" {
    binaryFilePath = filepath.Join("/private", binaryFilePath)
}
helper.Cmd("odo", append(args, "create", "--s2i", "java:8", cmpName, "--project",						commonVar.Project, "--binary", binaryFilePath...).ShouldPass()
```

Reply:
Thanks, This makes more sense